File: UniversityApiBackend/DTOs/Courses/CourseDetailDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using UniversityApiBackend.DTOs.Chapters;
using UniversityApiBackend.Models.DataModels;

namespace UniversityApiBackend.DTOs.Courses
{
    public class CourseDetailDTO
    {
        public int Id { get; set; }
        [Required, StringLength(50)]
        public string Name { get; set; } = string.Empty;

        [Required, StringLength(280)]
        public string ShortDescription { get; set; } = string.Empty;
        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        public string PublicGoal { get; set; } = string.Empty;

        public Level Level { get; set; } = Level.Basic;

        public ChapterDetailDTO Chapter { get; set; }
    }
}

File: Frontend/front-university/src/app/services/api/models/course-detail-dto.ts
/* tslint:disable */
/* eslint-disable */
import { Category } from './category';
import { ChapterDetailDto } from './chapter-detail-dto';
import { Level } from './level';
export interface CourseDetailDto {
  id:               number;
  name:             string;
  shortDescription: string;
  description:      string;
  publicGoal:       string;
  level:            Level;
  chapter:          ChapterDetailDto;
  categories:       Category[];
}




